### PR TITLE
lightning/br/dumpling: add session-token to external storage url

### PR DESCRIPTION
**Pull request by qiancai**
### First-time contributors' checklist 

- [ ] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs) that's required for repo owners to accept my contribution.

### What is changed, added or deleted? (Required)

add session-token to external storage url

Now we use the iam user with its access key id and secret access key to access s3. But the best practice from aws is using the iam role with its access ket id, secret access key and session token to access s3. Because the session token is time-limited.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions](https://github.com/pingcap/docs/blob/master/CONTRIBUTING.md#guideline-for-choosing-the-affected-versions).

- [x] master (the latest development version)
- [x] v6.6 (TiDB 6.6 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.4 (TiDB 6.4 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from: https://github.com/pingcap/docs-cn/pull/12947
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: br/backup-and-restore-storages.md [hunk 1]
```diff
@@ -156,7 +157,7 @@ It is recommended that you configure access to S3 using either of the following
 
 - Method 1: Specify the access key
 
-    If you specify an access key and a secret access key in the URL, authentication is performed using the specified access key and secret access key. Besides specifying the key in the URL, the following methods are also supported:
+    If you specify an access key and a secret access key in the URI, authentication is performed using the specified access key and secret access key. Besides specifying the key in the URI, the following methods are also supported:
```
Comment (qiancai) [inline br/backup-and-restore-storages.md:160]: Does the Chinese doc version need the same update?

File: br/backup-and-restore-storages.md [hunk 2]
```diff
@@ -175,7 +176,7 @@ It is recommended that you configure access to S3 using either of the following
 </div>
 <div label="GCS" value="gcs">
 
-You can configure the account used to access GCS by specifying the access key. If you specify the `credentials-file` parameter, the authentication is performed using the specified `credentials-file`. Besides specifying the key in the URL, the following methods are also supported:
+You can configure the account used to access GCS by specifying the access key. If you specify the `credentials-file` parameter, the authentication is performed using the specified `credentials-file`. Besides specifying the key in the URI, the following methods are also supported:
```
Comment (qiancai) [inline br/backup-and-restore-storages.md:179]: Does the Chinese doc version need the same update?

File: br/backup-and-restore-storages.md [hunk 3]
```diff
@@ -186,7 +187,7 @@ You can configure the account used to access GCS by specifying the access key. I
 
 - Method 1: Specify the access key
 
-    If you specify `account-name` and `account-key` in the URL, the authentication is performed using the specified access key and secret access key. Besides the method of specifying the key in the URL, BR can also read the key from the environment variable `$AZURE_STORAGE_KEY`.
+    If you specify `account-name` and `account-key` in the URI, the authentication is performed using the specified access key and secret access key. Besides the method of specifying the key in the URI, BR can also read the key from the environment variable `$AZURE_STORAGE_KEY`.
```
Comment (qiancai) [inline br/backup-and-restore-storages.md:191]: Does the Chinese doc version need the same update?

File: br/backup-and-restore-storages.md (hunk 1 above)
Comment (okJiang) [reply br/backup-and-restore-storages.md:160]: yes

File: br/backup-and-restore-storages.md (hunk 2 above)
Comment (okJiang) [reply br/backup-and-restore-storages.md:179]: yes

File: tidb-lightning/tidb-lightning-data-source.md [hunk 4]
```diff
@@ -334,29 +334,42 @@ type = '$3'
...
         -d 's3://my-bucket/test-data?role-arn=arn:aws:iam::888888888888:role/my-role'
     ```
 
+* Use access keys of an AWS IAM user to access S3 data:
+
+    ```bash
+    ./tidb-lightning --tidb-port=4000 --pd-urls=127.0.0.1:2379 --backend=local --sorted-kv-dir=/tmp/sorted-kvs \
+        -d 's3://my-bucket/test-data?access_key={my_access_key}&secret_access_key={my_secret_access_key}'
+
+* Use access keys of an AWS IAM role and session tokens to access S3 data:
```
Comment (okJiang) [inline tidb-lightning/tidb-lightning-data-source.md:366]:
```suggestion
* Use  the combination of AWS IAM role access keys and session tokens to access S3 data:
```

**Review by okJiang — APPROVED** [review]

**Review by hfxsd — APPROVED** [review]

File: tidb-lightning/tidb-lightning-data-source.md [hunk 5]
```diff
@@ -334,29 +334,42 @@ type = '$3'
...
+    ```bash
+    ./tidb-lightning --tidb-port=4000 --pd-urls=127.0.0.1:2379 --backend=local --sorted-kv-dir=/tmp/sorted-kvs \
+        -d 's3://my-bucket/test-data?access_key={my_access_key}&secret_access_key={my_secret_access_key}'
+
+* Use the combination of AWS IAM role access keys and session tokens to access S3 data:
+
+    ```bash
+    ./tidb-lightning --tidb-port=4000 --pd-urls=127.0.0.1:2379 --backend=local --sorted-kv-dir=/tmp/sorted-kvs \
+        -d 's3://my-bucket/test-data?access_key={my_access_key}&secret_access_key={my_secret_access_key}&session-token={my_session_token}'
+    ```
```
Comment (qiancai) [inline tidb-lightning/tidb-lightning-data-source.md:371]:
```suggestion
    ```bash
    ./tidb-lightning --tidb-port=4000 --pd-urls=127.0.0.1:2379 --backend=local --sorted-kv-dir=/tmp/sorted-kvs \
        -d 's3://my-bucket/test-data?access_key={my_access_key}&secret_access_key={my_secret_access_key}&session-token={my_session_token}'
    ```
```

File: tidb-lightning/tidb-lightning-data-source.md [hunk 6]
```diff
@@ -334,29 +334,42 @@ type = '$3'
...
     ```bash
     ./tidb-lightning --tidb-port=4000 --pd-urls=127.0.0.1:2379 --backend=local --sorted-kv-dir=/tmp/sorted-kvs \
         -d 's3://my-bucket/test-data?role-arn=arn:aws:iam::888888888888:role/my-role'
     ```
 
+* Use access keys of an AWS IAM user to access S3 data:
+
+    ```bash
+    ./tidb-lightning --tidb-port=4000 --pd-urls=127.0.0.1:2379 --backend=local --sorted-kv-dir=/tmp/sorted-kvs \
+        -d 's3://my-bucket/test-data?access_key={my_access_key}&secret_access_key={my_secret_access_key}'
```
Comment (qiancai) [inline tidb-lightning/tidb-lightning-data-source.md:364]:
```suggestion
        -d 's3://my-bucket/test-data?access_key={my_access_key}&secret_access_key={my_secret_access_key}'
    ```
```

**Pull request merged**